app/assets/javascripts/incident_reports/form.js

```diff
@@ -0,0 +1,46 @@
+$(document).on('turbolinks:load', () => {
+  if ($('#latlong-selector').length === 0) return;
+
+  let initialLatLng = [parseFloat($('#incident_latitude').val()), parseFloat($('#incident_longitude').val())];
+  if (isNaN(initialLatLng[0]) || isNaN(initialLatLng[1])) {
+    initialLatLng = [42.105552, -72.596511];
+  }
+
+  const map = L.map('latlong-selector').setView(initialLatLng, 13);
+  const tileLayer = L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+  });
+  const marker = L.marker(initialLatLng, {draggable: true});
+
+  const updateFields = () => {
+    const latLong = marker.getLatLng();
+    $('#incident_latitude').val(latLong.lat.toFixed(5));
+    $('#incident_longitude').val(latLong.lng.toFixed(5));
+
+    $.ajax({
+      url: 'https://nominatim.openstreetmap.org/reverse',
+      method: 'GET',
+      headers: {'User-Agent': 'PVTAIncidents/1.0'}, // play nice and identify ourselves
```

Comment:
I'll mention that the requirement is for a Referer _or_ a User-Agent. This is fine, but might not be strictly necessary.

Reply:
Changed my mind. If we're using it in development, I don't think `http://localhost:3000/` is tremendously useful to them in identifying who's abusing their service.